crates/host/src/wasmbus/mod.rs

```diff
@@ -1907,6 +1895,7 @@ impl Host {
             self.stop_actor(actor, host_id)
                 .await
                 .context("failed to stop old component")?;
+            // TODO: How to differentiate between old and new component if image reference is the same?
```

Comment:
An update should _only_ happen with separate image references. We should reject updates with the same image reference though, and that's a good issue to file if you're interested 😄 

Just like the above, I would recommend removing this comment unless it's tied to an issue

Reply:
I can take the initiative to check image references 👍

Reply:
Removed the line